src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.cs

```diff
@@ -16,14 +16,40 @@ internal partial class ClrThreadPool
         public static readonly ClrThreadPool ThreadPoolInstance = new ClrThreadPool();
 #pragma warning restore IDE1006 // Naming Styles
 
+        private const short MaxPossibleThreadCount = short.MaxValue;
+
+        private static short GetThreadCountConfig(string configName)
+        {
+            var value = AppContext.GetData(configName);
+            ushort numThreads = 0;
+            switch (value)
+            {
+                case string str:
+                    ushort.TryParse(str, out numThreads);
+                    break;
+                case short val:
```

Comment:
For non-string types, you could use Convert.ToInt32(object) and ignore all of the exceptions